### PR TITLE
WT-6250 Avoid passing reconcile struct to cell layer for stat logging

### PR DESCRIPTION
**Pull request by hbokhari**
This is my try at fixing the layer violation when passing WT_RECONCILE to low level cell packing code.

### CODE REVIEW

**Comment by hbokhari** [comment]
@agorrod, can you have a look at this. I have kept this in draft mode till we can agree on layering convention to follow.

**Comment by sulabhM** [comment]
What happened to WT-6220, there is an overlap between two and I thought we would come up with a final solution that merges all these validity window stats.

**Comment by hbokhari** [comment]
We believe stats here are orthogonal to what is being done in WT-6220 so the current decision was to keep them both and do a cleanup for layering violations.

**Comment by hbokhari** [comment]
@agorrod - This is another attempt on keeping the statistics. I have piggybacked on your changes to eliminate the need for passing around data structures.

I have also added three new statistics to track number of bytes used to write time window data on pages.

File: src/include/reconcile.h [hunk 1]
```diff
@@ -256,12 +256,15 @@ struct __wt_reconcile {
     WT_CURSOR_BTREE update_modify_cbt;
 
     /*
-     * Variables to track reconciled pages containing cells with time window values and prepared
+     * WT_REC_CELL_STATS--
```
Comment (agorrod) [inline src/include/reconcile.h:259]: I'm not averse to putting these into a structure, but I don't think it's necessary any more. I also don't think you need to give the structure a type (if you do, this doesn't look quite right).

File: dist/stat_data.py [hunk 2]
```diff
@@ -546,6 +546,8 @@ def __init__(self, name, desc, flags=''):
     RecStat('rec_time_aggr_oldest_start_ts', 'pages written including an aggregated oldest start timestamp '),
     RecStat('rec_time_aggr_oldest_start_txn', 'pages written including an aggregated oldest start transaction ID '),
     RecStat('rec_time_aggr_prepared', 'pages written including an aggregated prepare'),
+    RecStat('rec_time_window_bytes_ts', 'byte size of timestamps in pages written'),
+    RecStat('rec_time_window_bytes_txn', 'byte size of transaction IDs in pages written'),
```
[the remark this answers is not in the document]
Comment (hbokhari) [reply dist/stat_data.py:550]: Yes I realized that the numbers would not be accurate given we do bit of compression tricks in `__wt_vpack_uint()`. Let me see if we can do anything better or I'll update the description accordingly.

File: src/include/reconcile.h (hunk 1 above)
Comment (hbokhari) [reply src/include/reconcile.h]: Fixed

File: dist/stat_data.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hbokhari) [reply dist/stat_data.py]: I looked into getting more accurate results but unfortunately we can't do much without putting something in cell layer code. For now I have edited the description to highlight that these are approximate values.

**Pull request merged**